src/SFA.DAS.AANHub.Application/Attendances/Commands/PutAttendance/PutAttendanceCommandHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Common;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Entities;
+using SFA.DAS.AANHub.Domain.Interfaces;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using System.Text.Json;
+
+namespace SFA.DAS.AANHub.Application.Attendances.Commands.PutAttendance;
+
+public class PutAttendanceCommandHandler : IRequestHandler<PutAttendanceCommand, ValidatedResponse<PutCommandResult>>
+{
+    private readonly IAanDataContext _aanDataContext;
+    private readonly IAuditWriteRepository _auditWriteRepository;
+    private readonly IAttendancesWriteRepository _attendancesWriteRepository;
+                                       
+    public PutAttendanceCommandHandler(IAanDataContext aanDataContext,
+                                       IAuditWriteRepository auditWriteRepository,
+                                       IAttendancesWriteRepository attendancesWriteRepository)
+    {
+        _aanDataContext = aanDataContext;
+        _auditWriteRepository = auditWriteRepository;
+        _attendancesWriteRepository = attendancesWriteRepository;
+    }
+
+    public async Task<ValidatedResponse<PutCommandResult>> Handle(PutAttendanceCommand command, CancellationToken cancellationToken)
+    {
+        var existingAttendance = await _attendancesWriteRepository.GetAttendance(command.CalendarEventId, command.RequestedByMemberId);
+
+        if (existingAttendance != null)
+        {
+            if (existingAttendance.IsActive == command.RequestedActiveStatus)
+            {
+                return new ValidatedResponse<PutCommandResult>(new PutCommandResult(false));
+            }
+
+            await _attendancesWriteRepository.SetActiveStatus(command.CalendarEventId, command.RequestedByMemberId, command.RequestedActiveStatus);
```

Comment:
Because the entity is tracked, all you need to do here is set `existingAttendance.IsActive = command.IsAttending`. Your update method should look like:
```
void UpdateAttendance(Attendance existingAttendance, bool isAttending)
{
  var audit = new Audit {Action = "Put", Before =  JsonSerializer.Serialize(existingAttendance) ...};
  existingAttendance.isActive = isAttending;
  audit.After = JsonSerializer.Serialize(existingAttendance),
  _auditWriteRepository.Create(audit);  or  _dbContext.Audit.Add(audit)
  _dbContext.SaveChanges();
}